service/s3/s3manager/batch.go

```diff
@@ -43,6 +43,14 @@ type Error struct {
 	Key     *string
 }
 
+func newError(err error, bucket, key *string) Error {
+	return Error{
+		err,
+		bucket,
+		key,
+	}
+}
+
 func (err *Error) Error() string {
 	return err.OrigErr.Error()
```

Comment:
It would be helpful for the Error message to include the bucket and object key that the error failed on.